src/pages/index.js

```diff
@@ -109,10 +111,20 @@ addCardButton.addEventListener("click", handleAddCardFormButton);
 const user = new Userinfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
 function handleAddCardFormSubmit(data) {
-  renderCard({ name: data.title, link: data.link });
+  addCardPopup.creating(true);
+  api
+    .addCard(data.title, data.link)
+    .then((res) => {
+      renderCard(res);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => addCardPopup.creating(false));
   addCardPopup.close();
```

Comment:
Needs correction: Сlose modal in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again. (fix that for all the other popups too)

Reply:
Still needs correction